fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java

```diff
@@ -129,19 +127,28 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
             // in this case, CallOperator must have only one child ColumnRefOperator
             projectMap.put(entry.getKey(), entry.getValue().getChild(0));
         }
+
+        Map<ColumnRefOperator, ScalarOperator> newProjectMap = Maps.newHashMap();
+        ReplaceColumnRefRewriter rewriter = new ReplaceColumnRefRewriter(projectMap);
```

Comment:
what is the purpose rewrite by empty projectionMap?

Reply:
use projectMap to collect columnRefInfo and newProjectMap to save the rewriten results.

Reply:
Having clause or projection may contains the agg(col) reference. We should rewrite them use the rewriter.